crates/cubecl-core/src/frontend/element/atomic.rs

```diff
@@ -239,50 +195,84 @@ where
         new_var.into()
     }
 
-    fn __expand_max(
+    pub fn __expand_min(
         context: &mut CubeContext,
         pointer: <Self as CubeType>::ExpandType,
-        value: <Self::Primitive as CubeType>::ExpandType,
-    ) -> <Self::Primitive as CubeType>::ExpandType {
+        value: <Inner as CubeType>::ExpandType,
+    ) -> <Inner as CubeType>::ExpandType {
         let ptr: ExpandElement = pointer.into();
         let value: ExpandElement = value.into();
-        let new_var = context.create_local(Item::new(Self::Primitive::as_elem(context)));
+        let new_var = context.create_local(Item::new(Inner::as_elem(context)));
         context.register(Instruction::new(
-            AtomicOp::Max(BinaryOperator {
+            AtomicOp::Min(BinaryOperator {
                 lhs: *ptr,
                 rhs: *value,
             }),
             *new_var,
         ));
         new_var.into()
     }
+}
+
+/// An atomic type. Represents an shared value that can be operated on atomically.
```

Comment:
That doc should be on the top of the struct, not the impl block I think.

Reply:
That was an extra doc left over cause I just replaced the trait with the impl block. There's already documentation on the type, so I just removed it.